clog/readers.py

```diff
@@ -265,9 +266,13 @@ def __init__(self,
                  lines=None,
                  protocol_opts=None):
         if host is None or port is None:
-            primary_tail_host = random.choice(self.scribe_tail_services)
-            host = primary_tail_host['host']
-            port = primary_tail_host['port']
+            try:
+                primary_tail_host = random.choice(self.scribe_tail_services)
+                host = primary_tail_host['host']
+                port = primary_tail_host['port']
+            except (IndexError, ConfigurationError):
```

Comment:
`IndexError` would only happen if the configuration is specified but not it has the wrong format, right? I'm wondering if we should only catch `ConfigurationError`, and actually throw an exception otherwise.

Additionally, would it be possible to check if `self.scribe_tail_services` is `None` without relying on exceptions? There might be cases where the configuration is specified but with the wrong type, and with this change it would silently ignore them.

Reply:
Good point for the `IndexError`, I'll change it!

As for the second point, from my (low and bad) understanding of staticconf, this is all done lazily: the check that the configuration read is actually a list (or that the default value is actually a list) is done internally when using `self.scribe_tail_services` for the first time. We can't really easily catch it before. Happy to dig into more details if you need it, though :)

Reply:
Sorry, I confused `IndexError` with `KeyError`. I think this looks good.